### DISCUSSION TITLE
Representing list of dicts/models as list of strings aka flattening list of dicts/models into list of strings

### DISCUSSION THREAD

**Discussion by gbdlin** [pydantic/pydantic#8613] [Question]
It is possible in pydantic, with having given input data:

```json
{
  "name": "foo"
  "tags": [
    {
      "name": "tag1"
    },
    {
      "name": "tag2"
    }
  ]
}
```

 to have it serialized as follows:

```json
{
  "name": "foo"
  "tags": [
    "tag1",
    "tag2"
  ]
}
```

that is the list of tags is represented as list of strings, not as list of dictionaries as it was in the input data?

I've tried to use `AliasPath` for it, but it allows only for accessing a single value, not iterating over multiple ones.

My goal could be represented by this (not working) code:

```python
class Entry(BaseModel):
    name: str
    tags: Annotated[list[str], Field(validation_alias=AliasPath("tags", ..., "name"))]
```

**Comment by uriyyo** [comment]
Hi @gbdlin,

As an option you can define `Tag` class with custom serializer:
```py
from pydantic import BaseModel, model_serializer


class Tag(BaseModel):
    name: str

    @model_serializer(mode="plain", return_type=str)
    def serialize(self):
        return self.name


class Entry(BaseModel):
    name: str
    tags: list[Tag]
```

**Reply by gbdlin** [reply]
It's a solution, thank you. Unfortunately it requires creating a separate class for each field, which is not ideal, but thank you anyway.
